gprotorch/kernels/gnn_kernels/test_hu_iclr_2020.py

```diff
@@ -0,0 +1,357 @@
+        self.x_embedding2 = torch.nn.Embedding(
+            num_chirality_tag, self.embed_dim
+        )
+        torch.nn.init.xavier_uniform_(self.x_embedding1.weight.data)
+        torch.nn.init.xavier_uniform_(self.x_embedding2.weight.data)
+
+        # initialise GNN layers
+        self.gnns = torch.nn.ModuleList()
+        for layer in range(self.num_layers):
+            if gnn_type == "gin":
+                self.gnns.append(GINConv(emb_dim=self.embed_dim))
+            elif gnn_type == "gcn":
+                self.gnns.append(GCNConv(emb_dim=self.embed_dim))
+            else:
+                raise NotImplementedError("Invalid GNN layer type.")
+
+        # initialise BatchNorm layers
+        self.batch_norms = torch.nn.ModuleList()
+        for layer in range(self.num_layers):
+            self.batch_norms.append(torch.nn.BatchNorm1d(self.embed_dim))
+
+    def load_pretrained(self, pretrain_type, device):
+        """
+        Checks if a pretrained parameter set for the specified
+        pretraining procedure exists and updates the
+        network's state dict if it does..
+        Args:
+            pretrain_type: type of pretraining procedure
+            device: device to which to load the saved state dict
+
+        Returns:
+
+        """
+        pretrain_path = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            "pretrained_params",
+            f"hu_iclr_2020_{self.gnn_type}_{pretrain_type}.pth",
+        )
+        if not os.path.isfile(pretrain_path):
+            raise FileNotFoundError(
+                f"Specified GNN type ({self.gnn_type}) was not pretrained "
+                f"with requested approach ({pretrain_type})."
+            )
+        else:
+            pretrained_state_dict = torch.load(
+                pretrain_path, map_location=device
+            )
+            self.load_state_dict(pretrained_state_dict)
+
+    def forward(self, x, edge_index, edge_attr):
+        """
+        Forward function of the GNN class that takes a PyTorch geometric
+        representation of a molecule or a batch of molecules
+        and generates the node embeddings for each atom.
+        Args:
+            x: node features
+            edge_index: adjacency list
+            edge_attr: edge features
+
+        Returns: tensor of num_nodes x embedding_dim embeddings
+
+        """
+
+        # x[:, 0] corresponds to 'possible_atomic_num_list',
+        # x[:, 1] corresponds to 'possible_chirality_list'
+        x = self.x_embedding1(x[:, 0]) + self.x_embedding2(x[:, 1])
+
+        for layer in range(self.num_layers):
+            # x are atom features of the molecule and edge_attr the atomic features of the molecule
+            x = self.gnns[layer](x, edge_index, edge_attr)
+            x = self.batch_norms[layer](x)
+            if layer != self.num_layers - 1:
+                x = F.relu(x)
+
+        return x
+
+
+if __name__ == "__main__":
```

Comment:
Is this necessary? (I know it's included in the original repo, I was just wondering!)

Reply:
When I wrote this I did some quick testing directly from this script, but as we don't intend to run it as main we can take the main guard out.